### PR TITLE
Refactor/key oid candid alerts

### PR DESCRIPTION
**Pull request by AlxEnashi**
## Summary of the changes

Changed al uses of drop duplicates in detections and non detections dataframes by candid to by candid and oid.
Changed criteria for non detections insert in correctino step.

## Observations

<-- Add some notes to keep in mind !-->

#### Components that need updates and steps to follow

Pull from the changes to scribe and db. 
https://github.com/alercebroker/pipeline/pull/377

## About this PR:

- [ ] You added the necessary documentation for the team to understand your work.
- [ ] You linked this PR to the corresponding issue or user story.
- [ ] Your changes fulfill the Definition of Done of the issue or user story.
- [ ] Your changes were tested with manual testing before being submitted.
- [ ] Your changes were tested with automatic unit tests.
- [ ] Your changes were tested with automatic integration tests.

### CODE REVIEW

File: correction_step/correction/core/corrector.py [hunk 1]
```diff
@@ -26,11 +26,11 @@ def __init__(self, detections: list[dict]):
         """
         self.logger = logging.getLogger(f"alerce.{self.__class__.__name__}")
         self._detections = pd.DataFrame.from_records(detections, exclude={"extra_fields"})
-        self._detections = self._detections.drop_duplicates("candid").set_index("candid")
+        self._detections = self._detections.drop_duplicates(["candid", "oid"]).set_index("candid")
 
         self.__extras = {alert["candid"]: alert["extra_fields"] for alert in detections}
         extras = pd.DataFrame.from_dict(self.__extras, orient="index", columns=self._EXTRA_FIELDS)
-        extras = extras.reset_index(names=["candid"]).drop_duplicates("candid").set_index("candid")
+        extras = extras.reset_index(names=["candid"]).drop_duplicates(["candid", "oid"]).set_index("candid")
```
Comment (dirodriguezm) [inline correction_step/correction/core/corrector.py:33]: Esto es problemático, porque al setear index candid como ya no es único, si yo hago `df.loc[some_candid]`, va a retornar más de un elemento y romper el resto de la ejecución

**Review by dirodriguezm — CHANGES_REQUESTED** [review]

File: correction_step/correction/core/corrector.py (hunk 1 above)
Comment (AlxEnashi) [reply correction_step/correction/core/corrector.py:33]: ese extra son los extra fields? esos si deberian ser uno por candid no?

Comment (dirodriguezm) [reply correction_step/correction/core/corrector.py:33]: No solo para los extra fields. Cuando tienes una lista de detecciones con indice candid y usas `loc`, eventualmente puede retornar más de un valor

Comment (dirodriguezm) [reply correction_step/correction/core/corrector.py:33]: Lo que se me ocurre es hacer index por oid,candid pero desconozco el funcionamiento de loc con multiindex

**Comment by dirodriguezm** [comment]
Will merge this and the feature step will be fixed on a follow up PR

**Review by dirodriguezm — APPROVED** [review]

**Pull request merged**